util/formula_templater/shasums.go

```diff
@@ -0,0 +1,36 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+)
+
+func loadShasums(product string, version string) (map[string]string, error) {
+	shasums := make(map[string]string)
+	shasumURL := fmt.Sprintf("https://releases.hashicorp.com/%s/%s/%s_%s_SHA256SUMS", product, version, product, version)
+	resp, err := http.Get(shasumURL)
+	if err != nil {
+		return shasums, err
+	}
+	defer resp.Body.Close()
+	body, err := io.ReadAll(resp.Body)
+
+	lines := strings.Split(string(body), "\n")
+	for _, l := range lines {
+		if len(l) > 0 {
+			parts := strings.Split(l, "  ")
+			shasum := parts[0]
+			zip := parts[1]
```

Comment:
This could panic, it might be worth explicitly erroring if `len(parts) != 2` unless there's reasons I'm not seeing to be confident it will always be 2! (I know this is a machine-generated file, but humans are known to poke at things they shouldn't sometimes...)

Reply:
Good catch, updated